.github/workflows/ci.yaml

```diff
@@ -28,11 +28,27 @@ jobs:
         # - https://github.com/actions/virtual-environments/blob/main/images/macos/macos-13-Readme.md#xcode
         xcode-version: ['14.2', '14.1', '14.0.1', '13.4.1', '13.3.1', '13.2.1', '13.1']
     steps:
+      - name: Checkout the repo
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          path: ${{ env.WORKING_DIR }}
+
       - name: Update Go version using setup-go
         uses: actions/setup-go@0a12ed9d6a96ab950c8f026ed9f722fe0da7ef32 # v5.0.2
         if: matrix.go != 'tip'
         with:
           go-version: ${{ matrix.go }}
+          cache: true
+          cache-dependency-path: '**/go.sum'
+
+      - name: Install Go bootstrap compiler
+        uses: actions/setup-go@0a12ed9d6a96ab950c8f026ed9f722fe0da7ef32 # v5.0.2
+        if: matrix.go == 'tip'
+        with:
+          # Bootstrapping go tip requires 1.22.6
+          go-version: 1.22
+          cache: true
```

Comment:
Why are these cache manipulation needed?

May worth a comment.

Reply:
Without these we get errors from setup-go about it not being able to use the cache. (I think it might be related to having multiple go.sum files in our repo). I will add a comment.